### PR TITLE
Rename BioSample to Biosample

### LINKED ISSUE #684: Questionable use of snake_case (bio_sample_id instead of biosample_id)
The use of snake case here is confusing, since `BioSample` is a first class object. This probably comes from the confusing use of camelCase for the `BioSample` object name. Since we generally accept "biosample" as proper noun and use it as such in the documentation (this is also the case in major resources), I propose we change:
- `BioSample` => `Biosample`
- `bio_sample_id` => `biosample_id`

### PR DESCRIPTION
**Pull request by david4096**
This suggestion closes #684 by removing the capitalization from sample. This is a welcome change that better follows common conventions. Thanks @mbaudis !

### CODE REVIEW

**Comment by mbaudis** [comment]
+1 Obviously. Thanks @david4096!

This was one of the issues which had been discussed before, but probably succumbed to __branching madness__<sup>TM</sup>.

**Comment by dcolligan** [comment]
+1 

@david4096 you can probably rebase and merge this now... small change that's been open for a month with no objections

**Comment by david4096** [comment]
This will introduce breaking changes to the compliance suite, I believe. Let's wait to merge until we can verify.

File: src/main/proto/ga4gh/bio_metadata_service.proto [hunk 1]
```diff
@@ -39,10 +39,10 @@ service BioMetadataService {
       };
     };
 
-  // Gets a `BioSample` by ID.
-  // `GET /biosamples/{id}` will return a JSON version of `BioSample`.
-  rpc GetBioSample(GetBioSampleRequest)
-    returns (BioSample) {
+  // Gets a `Biosample` by ID.
+  // `GET /biosamples/{id}` will return a JSON version of `Biosample`.
+  rpc GetBioSample(GetBiosampleRequest)
```
Comment (ejacox) [inline src/main/proto/ga4gh/bio_metadata_service.proto]: Line 44, change GetBioSample to GetBiosample

File: src/main/proto/ga4gh/bio_metadata_service.proto [hunk 2]
```diff
@@ -39,10 +39,10 @@ service BioMetadataService {
...
+    returns (Biosample) {
         option (google.api.http) = {
           get: "/v0.6.0a8/biosamples/{bio_sample_id}"
```
Comment (ejacox) [inline src/main/proto/ga4gh/bio_metadata_service.proto]: Change bio_sample_id to biosample_id

**Comment by ejacox** [comment]
Pull doc/source/metadata_schema.png from my branch (biosample-1394, commit 0b2745f) to get the updated png file.

**Comment by david4096** [comment]
Thanks @ejacox got it!

**Comment by david4096** [comment]
@ejacox updated

**Pull request merged**